Broken ci for aarch64 windows build

See logs here: https://github.com/DenuxPlays/dyncloud/actions/runs/12304496065/job/34342058658

Is it possible to make ring compile at aarch64 windows.
Here are the current logs:

![image](https://github.com/user-attachments/assets/06faa836-1d21-4ae5-8e29-b07a577f2a71)

![image](https://github.com/user-attachments/assets/ca031544-0792-45f9-bf36-ca92e31f9eb4)

In text form:
```
error: failed to run custom build command for `ring v0.17.8`

Caused by:
  process didn't exit successfully: `/__w/dyncloud/dyncloud/target/dist/build/ring-[52](https://github.com/DenuxPlays/dyncloud/actions/runs/12305224895/job/34344358806#step:9:53)4ab4a82563183a/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=RING_PREGENERATE_ASM
  cargo:rustc-env=RING_CORE_PREFIX=ring_core_0_17_8_
  OPT_LEVEL = Some(3)
  OUT_DIR = Some(/__w/dyncloud/dyncloud/target/aarch64-pc-windows-msvc/dist/build/ring-bdcbb9dd337[53](https://github.com/DenuxPlays/dyncloud/actions/runs/12305224895/job/34344358806#step:9:54)af3/out)
  TARGET = Some(aarch64-pc-windows-msvc)
  cargo:rerun-if-env-changed=VCINSTALLDIR
  VCINSTALLDIR = None
  HOST = Some(x86_64-unknown-linux-gnu)
  cargo:rerun-if-env-changed=CC_aarch64-pc-windows-msvc
  CC_aarch64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_aarch64_pc_windows_msvc
  CC_aarch64_pc_windows_msvc = Some(clang-cl)
  cargo:rerun-if-env-changed=CC_KNOWN_WRAPPER_CUSTOM
  CC_KNOWN_WRAPPER_CUSTOM = None
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some(crt-static,neon)
  DEBUG = Some(false)
  cargo:rerun-if-env-changed=CFLAGS_aarch64-pc-windows-msvc
  CFLAGS_aarch64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_pc_windows_msvc
  CFLAGS_aarch64_pc_windows_msvc = Some(--target=aarch64-pc-windows-msvc -Wno-unused-command-line-argument -fuse-ld=lld-link /imsvc/github/home/.cache/cargo-xwin/xwin/crt/include /imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/ucrt /imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/um /imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/shared  )
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  CARGO_ENCODED_RUSTFLAGS = Some(-Ctarget-feature=+crt-static-Clinker-flavor=lld-link-Lnative=/github/home/.cache/cargo-xwin/xwin/crt/lib/aarch64-Lnative=/github/home/.cache/cargo-xwin/xwin/sdk/lib/um/aarch64-Lnative=/github/home/.cache/cargo-xwin/xwin/sdk/lib/ucrt/aarch64)

  --- stderr
  running LC_ALL="C" "clang" "-O3" "-ffunction-sections" "-fdata-sections" "--target=aarch64-pc-windows-msvc" "-I" "include" "-I" "/__w/dyncloud/dyncloud/target/aarch64-pc-windows-msvc/dist/build/ring-bdcbb9dd33753af3/out" "--target=aarch64-pc-windows-msvc" "-Wno-unused-command-line-argument" "-fuse-ld=lld-link" "/imsvc/github/home/.cache/cargo-xwin/xwin/crt/include" "/imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/ucrt" "/imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/um" "/imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/shared" "-fvisibility=hidden" "-std=c1x" "-Wall" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-g3" "-DNDEBUG" "-c" "-o/__w/dyncloud/dyncloud/target/aarch64-pc-windows-msvc/dist/build/ring-bdcbb9dd33753af3/out/aesv8-armx-win64.o" "/usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.8/pregenerated/aesv8-armx-win64.S"
  clang: error: no such file or directory: '/imsvc/github/home/.cache/cargo-xwin/xwin/crt/include'
  clang: error: no such file or directory: '/imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/ucrt'
  clang: error: no such file or directory: '/imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/um'
  clang: error: no such file or directory: '/imsvc/github/home/.cache/cargo-xwin/xwin/sdk/include/shared'
  thread 'main' panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.8/build.rs:670:9:
  execution failed
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...

  × failed to find bin dyncloud.exe for path+file:///__w/dyncloud/dyncloud#1.5.0
  help: did the above build fail?

Error: Process completed with exit code 2[55](https://github.com/DenuxPlays/dyncloud/actions/runs/12305224895/job/34344358806#step:9:56).
```

--- a/cargo-dist/src/backend/ci/github.rs
+++ b/cargo-dist/src/backend/ci/github.rs
@@ -8,10 +8,10 @@ use axoasset::{LocalAsset, SourceFile};
 use axoprocess::Cmd;
 use camino::{Utf8Path, Utf8PathBuf};
 use cargo_dist_schema::{
-    target_lexicon::{self, OperatingSystem, Triple},
-    AptPackageName, ChocolateyPackageName, GhaRunStep, GithubGlobalJobConfig, GithubLocalJobConfig,
-    GithubMatrix, GithubRunnerConfig, GithubRunnerRef, GithubRunners, HomebrewPackageName,
-    PackageInstallScript, PackageVersion, PipPackageName, TripleNameRef,
+    target_lexicon::{self, Architecture, OperatingSystem, Triple},
+    AptPackageName, ChocolateyPackageName, ContainerImageRef, GhaRunStep, GithubGlobalJobConfig,
+    GithubLocalJobConfig, GithubMatrix, GithubRunnerConfig, GithubRunnerRef, GithubRunners,
+    HomebrewPackageName, PackageInstallScript, PackageVersion, PipPackageName, TripleNameRef,
 };
 use itertools::Itertools;
 use serde::{Deserialize, Serialize};
@@ -26,7 +26,7 @@ use crate::{
         JinjaGithubRepoPair, JobStyle, ProductionMode, PublishStyle, SystemDependencies,
     },
     errors::DistResult,
-    platform::github_runners::target_for_github_runner_or_default,
+    platform::{github_runners::target_for_github_runner_or_default, targets},
     CargoBuildWrapper, DistError, DistGraph, SortedMap, SortedSet,
 };
 
@@ -661,11 +661,30 @@ fn github_runner_for_target(
     let result = Some(match target_triple.operating_system {
         OperatingSystem::Linux => runner_to_config(GithubRunnerRef::from_str("ubuntu-20.04")),
         OperatingSystem::Darwin => runner_to_config(GithubRunnerRef::from_str("macos-13")),
-        OperatingSystem::Windows => runner_to_config(GithubRunnerRef::from_str("windows-2019")),
+        OperatingSystem::Windows => {
+            // Default to cargo-xwin for Windows cross-compiles
+            if target_triple.architecture != Architecture::X86_64 {
+                cargo_xwin()
+            } else {
+                runner_to_config(GithubRunnerRef::from_str("windows-2019"))
+            }
+        }
         _ => return Ok(None),
     });
 
     Ok(result)
+}
+
+fn cargo_xwin() -> GithubRunnerConfig {
+    GithubRunnerConfig {
+        runner: GithubRunnerRef::from_str("ubuntu-20.04").to_owned(),
+        host: targets::TARGET_X64_LINUX_GNU.to_owned(),
+        container: Some(cargo_dist_schema::ContainerConfig {
+            image: ContainerImageRef::from_str("messense/cargo-xwin").to_owned(),
+            host: targets::TARGET_X64_LINUX_MUSL.to_owned(),
+            package_manager: None,
+        }),
+    }
 }
 
 fn brewfile_from<'a>(packages: impl Iterator<Item = &'a HomebrewPackageName>) -> String {

--- a/cargo-dist/src/tasks.rs
+++ b/cargo-dist/src/tasks.rs
@@ -486,10 +486,6 @@ pub fn build_wrapper_for_cross(
         },
         // compiling for Windows (making PE binaries, .dll files, etc.)
         OperatingSystem::Windows => match host.operating_system {
-            OperatingSystem::Windows => {
-                // this is just cross-arch, hopefully no wrappers are needed?
-                Ok(Some(CargoBuildWrapper::ZigBuild))
-            }
             OperatingSystem::Linux | OperatingSystem::Darwin => {
                 // cargo-xwin is made for that
                 Ok(Some(CargoBuildWrapper::Xwin))
@@ -498,7 +494,7 @@ pub fn build_wrapper_for_cross(
                 Err(DistError::UnsupportedCrossCompile {
                     host: host.clone(),
                     target: target.clone(),
-                    details: format!("no idea how to cross-compile from {host} to windows"),
+                    details: format!("no idea how to cross-compile from {host} to windows with architecture {}", target.architecture),
                 })
             }
         },

--- a/cargo-dist/tests/integration-tests.rs
+++ b/cargo-dist/tests/integration-tests.rs
@@ -1997,7 +1997,7 @@ fn axolotlsay_cross2() -> Result<(), miette::Report> {
 [workspace.metadata.dist]
 cargo-dist-version = "{dist_version}"
 installers = ["shell", "powershell"]
-targets = ["aarch64-unknown-linux-gnu"]
+targets = ["aarch64-unknown-linux-gnu", "aarch64-pc-windows-msvc"]
 ci = ["github"]
 unix-archive = ".tar.gz"
 windows-archive = ".tar.gz"

